dora/src/bytecode/dumper.rs

```diff
@@ -920,6 +920,13 @@ impl<'a> BytecodeVisitor for BytecodeDumper<'a> {
         self.emit_new_array("NewArray", dest, cls, length);
     }
 
+    fn visit_array_length(&mut self, dest: Register, arr: Register) {
+        self.emit_reg2("ArrayLength", dest, arr);
+    }
+    fn visit_array_bound_check(&mut self, arr: Register) {
```

Comment:
I think this needs a second register for the index.

Reply:
Of course, don't know how i missed that.